### PR TITLE
ESBJAVA-3480 - global keep alive setting support was added.

### PR DESCRIPTION
**Pull request by ravindraranwala**

### CODE REVIEW

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java [hunk 1]
```diff
@@ -140,11 +140,16 @@ public static TargetRequest create(MessageContext msgContext,
...
+			// keep alive in synapse config
+			final String noKeepAlive =
+			                           (String) msgContext.getProperty(PassThroughConstants.NO_KEEPALIVE);
+
+			// Global Keep Alive settings in passthru props.
+			final boolean noGlobalKeepalive = configuration.isDisableGlobalKeepalive();
+
+			if (("true".equals(noKeepAlive) || noGlobalKeepalive) && (!"false".equals(noKeepAlive))) {
+				request.setKeepAlive(false);
+			}
```
Comment (Buddhima) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/TargetRequestFactory.java]: Please fix the indentation issue here

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/BaseConfiguration.java [hunk 2]
```diff
@@ -87,6 +88,8 @@ public void build() throws AxisFault {
         ioReactorConfig = buildIOReactorConfig();
 
         bufferFactory = new BufferFactory(iOBufferSize, new HeapByteBufferAllocator(), 512);
+		// Setting up the GlobalKeepalive property value here.
+		disableGlobalKeepalive = conf.isKeepAliveDisabled();
```
Comment (Buddhima) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/BaseConfiguration.java]: Please fix the indentation issue here

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/BaseConfiguration.java [hunk 3]
```diff
@@ -103,19 +105,23 @@ public ConfigurationContext getConfigurationContext() {
...
+                                                   8 * 1024))
+              .setParameter(HttpProtocolParams.ORIGIN_SERVER,
+                            conf.getStringProperty(HttpProtocolParams.ORIGIN_SERVER,
+                                                   "WSO2-PassThrough-HTTP"))
+              .setParameter(HttpProtocolParams.USER_AGENT,
+                            conf.getStringProperty(HttpProtocolParams.USER_AGENT,
+                                                   "Synapse-PT-HttpComponents-NIO"));
+        // setParameter(HttpProtocolParams.HTTP_ELEMENT_CHARSET,
+        // conf.getStringProperty(HttpProtocolParams.HTTP_ELEMENT_CHARSET,
+        // HTTP.DEFAULT_PROTOCOL_CHARSET));//TODO:This does not works with
```
Comment (prabathabey) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/BaseConfiguration.java]: Fix TODOs.

**Comment by ravindraranwala** [comment]
Oboleted pr.

**Pull request closed without merging**